docs/project/requirements/architecture.md

```diff
@@ -0,0 +1,147 @@
+# Architecture
+
+Un schéma d'implémentation détaillé sera établi pour décrire l'architecture globale de l'infrastructure, y compris les différents composants, leur interaction et leur déploiement.
+
+
+## Application
+
+
+### Proxy: Traefik
+
+* Application router
+* High-availability by fault tolerance
+* Load-balancing by requests distribution
+  ALB ?
+* Helm chart
+* Version
+
+
+### Application: FastAPI
+
+* Python
+  * Version:
+  * Packaging
+* Project management? Poetry?
+
+
+### Databases: PostgreSQL
+
+* Helm chart
+* Version
+* Master/slave for high availability and read load balancing
+* Master failover
+
+
+## Infrastructure
+
+
+### Network
+
+
+#### Application Load Balancer (ALB)
+
+
+#### Virtual Private Cloud (VPC)
+
+* VPC subnet
+
+
+#### Internet Gateways (IGW)
+
+* Global egress
+
+
+#### Availability Zones (AZ)
+
+* 2 availability zones for high availability
+
+
+#### NAT Gateways (NGW)
+
+* Per AZ egress
+* In public subnet
+
+
+#### Subnets
+
+
+##### Public subnets
+
+* Application access
+* Bastion access
+* NAT gateways ?
+
+
+##### Private subnets
+
+* Database
+* Admin
+
+
+### Compute
+
+#### Nodes
+
+* EC2 compute nodes
+  * Application nodes
+  * Database nodes
+  * Bastion node
+* Auto-scaling groups
+
+
+#### Orchestration: Kubernetes
+
+#### Pods
+
+##### ReplicaSets
+
+* Traefik
+* Application
+* Database
+* cert-manager
+* Metrics server
+* bastion
+
+##### DaemonSets
+
+* Any ?
+
+
+### Storage
+
+#### CSI
+
+##### EBS
+
+* FIXME
+
+
+## CI/CD
+
+### GitHub Actions
+
+
+#### Actions
+
+#### Workflows
+
+* Application
+* Infrastructure
+
+
+## Observability
+
+### Log management (ELK/EFK)
```

Comment:
We don't use ELK right now.

Reply:
Do we have some sort of log aggregation and management? Loki? Graylog? Bare rsyslog aggregation to start with?

Reply:
Prometheus pull the metrics logs and grafana use is database to display in a dashbord the datas. 

In other case, in a developpment and preprod environement K9S is used to read the logs.

We used also the dashboard to follow our budget who is limited to 80 USD for all the project.

Reply:
This is missing right now. I'll look into it.

Reply:
Loki looks like a nice & cool NKOTB, but the fact that it doesn't indexes logs makes it a one of a kind I need to learn to know...
I tend to favor EFK over ELK because Fluentd and Fluentbit look lighter and leaner than that fat Java stash.
Finally, Graylog has always looked like a nice integrated, batteries included, solution.